src/main/scala/quipu/Interpreter.scala

```diff
@@ -52,7 +53,7 @@ object Interpreter {
         knots.head match {
           case ReferenceKnot =>
             val ref = stack(0) match {
-              case i: Int if (i < code.length) => code(i)(0) match {
+              case i: BigInt if (i < code.length) => code(i.intValue)(0) match {
```

Comment:
Is it guaranteed that the `BigInt` will be small enough to call `intValue`?

Reply:
I think so, because `code.length` returns an Int, so if `i` is less than `code.length`, it should be convertible to Int. Though it just occurred to me that there could be a problem if `i` is negative...